pontoon/base/static/js/translate.js

```diff
@@ -1998,44 +2002,128 @@ var Pontoon = (function (my) {
     /*
      * Open first entity in the entity list in the editor
      */
-    openFirstEntity: function () {
+    openFirstEntity: function() {
       $('#entitylist .entity:first').mouseover().click();
     },
 
+    /*
+     * Opens entity with the given id.
+     */
+    openEntity: function(entityId) {
+      $('#entitylist .entity[data-entry-pk=' + entityId + ']').mouseover().click();
+    },
 
     /*
      * Create user interface
      */
     createUI: function () {
+      var self = this;
       // Show message if provided
       if ($('.notification li').length) {
         $('.notification').css('opacity', 100).removeClass('hide');
       }
 
-      this.setMainLoading(false);
-      this.toggleInplaceElements();
-      this.resetColumnsWidth();
-      this.updateMainMenu();
-      this.updateProjectInfo();
-      this.updateProfileMenu();
-      this.updateSaveButtons();
-      this.renderEntityList();
+      self.setMainLoading(false);
+      self.toggleInplaceElements();
+      self.resetColumnsWidth();
+      self.updateMainMenu();
+      self.updateProjectInfo();
+      self.updateProfileMenu();
+      self.updateSaveButtons();
+      self.renderEntityList();
 
-      this.updateProgress();
+      self.updateProgress();
       $("#progress").show();
 
       $("#project-load").hide();
 
-      // If 2-column layout opened by default, open first entity in the editor
-      if (this.app.advanced) {
-        this.openFirstEntity();
+      if (self.requiresInplaceEditor()) {
+        self.waitForAttribute('entities').then(function() {
```

Comment:
I guess that some other problem caused that entities were empty at that point, so i decided to wait for them. I think we can remove this wait